packages/dashboard/src/layout/LayoutUtils.ts

```diff
@@ -248,7 +253,7 @@ class LayoutUtils {
     if (!stack && matchComponentType) {
       stack = this.getStackForConfig(
         root,
-        { component: config.component },
+        { component: (config as { component?: string }).component },
```

Comment:
This is kind of strange, but seems messier to do it a different way... Could create a type predicate just for this, but seems kind of messy.

Reply:
Yeah, there's a limitation of TypeScript for scenarios where a type is changed as a result of setting something.